can/interfaces/socketcand/socketcand.py

```diff
@@ -13,12 +13,96 @@
 import socket
 import time
 import traceback
+import urllib.parse as urlparselib
+import xml.etree.ElementTree as ET
 from collections import deque
+from typing import List
 
 import can
 
 log = logging.getLogger(__name__)
 
+DEFAULT_SOCKETCAND_DISCOVERY_ADDRESS = ""
+DEFAULT_SOCKETCAND_DISCOVERY_PORT = 42000
+
+
+def detect_beacon():
+    sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    sock.bind((DEFAULT_SOCKETCAND_DISCOVERY_ADDRESS, DEFAULT_SOCKETCAND_DISCOVERY_PORT))
+    log.info(
+        f"Listening on for socketcand UDP advertisement on {DEFAULT_SOCKETCAND_DISCOVERY_ADDRESS}:{DEFAULT_SOCKETCAND_DISCOVERY_PORT}"
+    )
+
+    # Time between beacons no more than 3 seconds. Allow for at least 3
```

Comment:
One timeout should be sufficient for all beacons, why do you multiply the timeout?

Reply:
Beacons are send via UDP broadcast and are not guaranteed delivery. The time between each beacon broadcast is 3 seconds. So - 12 sec would be required to guarantee a receiving window of at least 3 beacons.

Reply:
Aren't the beacons indepedent of each other? I thought the max. time between broadcasts of each individual beacon is 3s.

Reply:
You are correct. I think there is a misunderstanding. When I mentioned beacon, I meant beacon-packet. Assume there is a single socketcand server on the network. I proposed a timeout that would guarantee a receiving window of at least 2 beacon packets. I didn't want to set a timeout on which only a single beacon packet could be received - as UDP does not guarantee delivery.

(The above would be true if there were multiple socketcand servers as well, as they send their beacon packets independently, with 3s between beacon packet broadcasts).

Reply:
But packet loss is probably an exception, i would not wait this long per default. A little over 3 seconds should be enough unless the the user chooses to wait longer.

Reply:
Agreed.